### ISSUE TITLE
Incorrect URL created when qBittorrent is behind a reverse proxy causing 404 error

### ISSUE THREAD

**Issue by zachoooo** [rmartin16/qbittorrent-api#37]
I have my qBittorrent instance on a docker network along with a reverse proxy and the Web UI is available at http://[HOST]/qBittorrent/. The API is available and works correctly, when using curl or my browser. qbittorrent-api seems to ignore the base url and so I've been getting 404s when trying to use the API. Here is the example code:
```
qbt_client = qbittorrentapi.Client(host='http://[HOST]/qBittorrent/', username='[USERNAME]', password='[PASSWORD]')
```

**Comment by rmartin16** [comment]
can you please provide one of the working `curl` statements?

it looks like you're prefixing the Web API paths with `qBittorrent`....presumably the reverse proxy is then trimming the prefix to the qBittorrent instance. if that's the case, then yes...that is not supported....right now anyway. only the "host" portion (and a port) of the string provided for `host` is [actually used](https://github.com/rmartin16/qbittorrent-api/blob/master/qbittorrentapi/request.py#L395). the absolute paths for the API are hardcoded.

**Comment by zachoooo** [comment]
Yes my reverse proxy is trimming the prefixes. qBittorrent currently doesn't support base urls, so I have to trim it to make it available through the reverse proxy. There is an issue open for it on qBittorrent: https://github.com/qbittorrent/qBittorrent/issues/5693

It has been open for quite some time though, so I don't know when that feature will be added. It's possible that this could be considered an upstream issue. Anyways here is the command:

```bash
curl http://192.168.1.69/qBittorrent/api/v2/app/version --cookie "[BROWSER COOKIE]"
```

I copied the cookie from my browser after logging in through the Web UI. I didn't want to include it in case it contained any sensitive information, but the output does work.

```bash
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100     6  100     6    0     0   6000      0 --:--:-- --:--:-- --:--:--  6000v4.3.1
```

**Comment by rmartin16** [comment]
hmm....even if they supported a base url, i still think this client would be broken....since it'll always send the default paths....unless qbittorrent just accepted either i guess.

in theory, it wouldn't be too hard to prefix the API path with any user provided base path. just a little concerned that could be a breaking change to some users if they are randomly including a path in their instantiations right now....seems pretty unlikely though.

i can at least mock something up in the next few days. thanks for letting me know.

**Comment by zachoooo** [comment]
I guess this is probably better addressed upstream by qBittorrent devs. I can always just use Docker DNS to connect to the API endpoints directly instead of using a reverse proxy. Thanks for looking at it so quickly though!

**Comment by rmartin16** [comment]
@zachoooo, I released v2020.12.14. It should now support your instantiation below and make calls using `/qBittorrent` at the base of the path. Please let me know if you run in to any problems. Thanks.

```python
qbt_client = qbittorrentapi.Client(host='http://[HOST]/qBittorrent/', username='[USERNAME]', password='[PASSWORD]')
```
